File: backend-api/src/main/java/vn/aptech/backendapi/service/Doctor/DoctorService.java
package vn.aptech.backendapi.service.Doctor;

import vn.aptech.backendapi.dto.CustomDoctorForEdit;
import vn.aptech.backendapi.dto.DoctorDto;

import java.util.List;
import java.util.Optional;

public interface DoctorService {
    List<DoctorDto> findAll();
    Optional<DoctorDto> findById(int id);
    Optional<DoctorDto> findByUserId1(int userId);
    Optional<DoctorDto> findByUserId(int userId);
    //Hien Create 30/4/2024
    List<DoctorDto> findDoctorsByDepartmentId(int departmentId);
    // writed by An in 5/11
    DoctorDto updatePriceAndDepartment(int id , double price , int departmentId);
    List<DoctorDto> findAllWithAllStatus();
    List<DoctorDto> findDoctorsByDepartmentIdWithAllStatus(int departmentId);

    List<DoctorDto> searchDoctorsByName(String name);

    // Hien - 28/5/2024
    DoctorDto update(int id, DoctorDto doctorDto);

    boolean deleteDoctorFromDepartment(int doctorId);

    List<DoctorDto> doctorNotDepartment();

    boolean addDoctorToDepartment(int departmentId , int doctorId);

    CustomDoctorForEdit getDoctorDetail(int doctorId);
    DoctorDto editDoctor(int doctorId , CustomDoctorForEdit doctor);

}

File: bookingdoctor_web/src/services/API/doctorService.js
import * as request from "../../ultils/request";

export const getAllDoctor = async () => {
    try {
        const response = await request.get('doctor/all');
        return response
    } catch (error) {
        console.log(error);
        throw error;
    }
};

export const getAllDoctorWithStatus = async () => {
    try {
        const response = await request.get('doctor/allwithallstatus');
        return response
    } catch (error) {
        console.log(error);
        throw error;
    }
};


export const detailDoctor = async (id) => {
    try {
        const response = await request.get(`doctor/${id}`);
        return response;
    } catch (error) {
        console.log(error);
        throw error;
    }
}

export const findDoctorByUserId = async (id) => {
    try {
        const response = await request.get(`doctor/findbyuserid/${id}`);
        return response;
    } catch (error) {
        console.log(error);
        throw error;
    }
}



export const updateDoctor = async (id, price, departmentid) => {
    try {
        const response = await request.put(`doctor/updatepriceanddepartment/${id}/${price}/${departmentid}`);
        return response
    } catch (error) {
        console.log(error);
        throw error;
    }
};

export const getDoctorNotDepartment = async () => {
    try {
        const response = await request.get(`doctor/doctornotdepartment`);
        return response;
    } catch (error) {
        console.log(error);
        throw error;
    }
}


export const addDoctorToDepartment = async (doctorId,departmentId) => {
    try {
        const response = await request.put(`doctor/adddoctortodepartment/${doctorId}/${departmentId}`);
        return response
    } catch (error) {
        console.log(error);
        throw error;
    }
};